msnoise/s000installer.py

```diff
@@ -124,59 +184,32 @@ def main(tech=None, hostname="localhost", username="msnoise",
         password = None
         hostname = filename
     else:
-        engine = create_engine('mysql+pymysql://%s:%s@%s/%s' % (username,
-                                                                password,
-                                                                hostname,
-                                                                database),
+        engine = create_engine('mysql+pymysql://%s:%s@%s/%s'
+                               % (username, password, hostname, database),
                                echo=False)
 
-    # from .api import create_database_inifile
     create_database_inifile(tech, hostname, database, username, password,
                             prefix)
 
-    from .msnoise_table_def import PrefixerBase, Config
-    PrefixerBase._the_prefix = prefix
-    # create tables
-    PrefixerBase.metadata.create_all(engine)
+    schema = declare_tables(prefix)
+    schema.Base.metadata.create_all(engine)
+
     Session = sessionmaker(bind=engine)
     session = Session()
-    
+
+    # Add default configuration values to the database
     for name in default.keys():
-        session.add(Config(name=name, value=default[name][1]))
-    
-    try:
-        session.commit()
-    except IntegrityError:
-        print("The database seems to already exist and is not empty, cannot"
-              " continue")
-        return "Integrity Error - DB already exists"
-    if prefix != "":
-        prefix = prefix + "_"
-    # TODO move those calls to a def and call it from install / msnoise.scripts
-    try:
-        session.execute("CREATE UNIQUE INDEX job_index ON %sjobs (day, pair, "
-                        "jobtype)" % prefix)
-        session.commit()
-    except:
-        logging.info("It looks like the v1.5 'job_index' is already in the DB")
-        session.rollback()
+        session.add(schema.Config(name=name, value=default[name][1]))
 
     try:
-        session.execute("CREATE INDEX job_index2 ON %sjobs (jobtype, flag)"
-                        % prefix)
         session.commit()
-    except:
-        logging.info("It looks like the v1.6 'job_index2' is already in the DB")
-        session.rollback()
+    except IntegrityError:
+        logging.error("The database seems to already exist and is not empty, "
+                      "cannot continue")
+        return 1
 
-    try:
-        session.execute("CREATE UNIQUE INDEX da_index ON %sdata_availability ("
-                        "path, file, net, sta, comp)" % prefix)
-        session.commit()
-    except:
-        logging.info("It looks like the v1.5 'da_index' is already in the DB")
-        session.rollback()
+    create_indices(session)
```

Comment:
no need to pass ``prefix`` here?

Reply:
Correct, we should pass prefix to be able to correctly create the indices.